xdsl/dialects/x86/ops.py

```diff
@@ -1195,6 +1191,65 @@ def custom_print_attributes(self, printer: Printer) -> Set[str]:
         return {"offset"}
 
 
+@irdl_op_definition
+class M_PopOp(IRDLOperation, X86Instruction, ABC):
+    """
+    Copies the value at the top of the stack into [r1] and increases %rsp.
+    https://www.felixcloutier.com/x86/pop
+    """
+
+    name = "x86.m.pop"
+
+    rsp_input = operand_def(GeneralRegisterType("rsp"))
+    destination = operand_def(
+        R1InvT
```

Comment:
This seems off, shouldn't it be a Generic parameter?

Reply:
Not sure what you mean? I define operands the same way in all the other instructions inheriting directly from IRDLOperation. It wouldn't let me pass Generic arguments iirc.

Reply:
Hmm, not sure how I missed this. This is a TypeVar, it's designed to be passed to `Generic` at the type level. When used like this, it's not any different to just passing the `bound` of the TypeVar, and possibly technically invalid Python, in my understanding. I'd say we should either make the base classes generic in their register types, or switch the operand constraints to pass the bound instead.